crates/astria-composer/src/executor/mod.rs

```diff
@@ -199,6 +203,16 @@ impl Executor {
     /// An error is returned if connecting to the sequencer fails.
     #[instrument(skip_all, fields(address = %self.address))]
     pub(super) async fn run_until_stopped(mut self) -> eyre::Result<()> {
+        let remote_chain_id = self
```

Comment:
IMO, factor this into a method `async fn setup` or `async fn init` or `async fn pre_run_checks` and perform the checks there.

From inside `run_until_stopped` you can then do something like below so that your task is a good citizen and doesn't just ignore the shutdown signal during setup.


```rust
select!(
    biased;
    () = self.shutdown_token.cancelled() => {
        info!("received shutdown signal while running initialization routines; exiting");
        return Ok(());
    }

    res = self.setup() => {
        if let Err(e) = res {
            bail!("runtime initialization failed");
        }
);
```

Reply:
Implemented in 8523e04.